homeassistant/components/onkyo/media_player.py

```diff
@@ -316,21 +348,18 @@ class OnkyoMediaPlayer(MediaPlayerEntity):
     def __init__(
         self,
         receiver: Receiver,
-        sources: dict[str, str],
         zone: str,
-        max_volume: int,
-        volume_resolution: int,
+        *,
+        volume_resolution: VolumeResolution,
+        max_volume: float,
+        sources: dict[InputSource, str],
     ) -> None:
         """Initialize the Onkyo Receiver."""
         self._receiver = receiver
-        name = receiver.name
+        name = receiver.model_name
```

Comment:
What was the `.name` before?

Reply:
iiirc use defined name. The model name comes from the device itself.

Reply:
It depends on how the integration was setup.

If it was defined host, then name was declared there as well. If it was discovery, then the name was the same as model name, so nothing changes.

That said, after I added `unique_id` I don't think it is a problem. If some users are not happy with the name, they can always change it themselves. And if they already did, it is stored in the registry, so this doesn't have any impact.

Reply:
Agree. Also, I think it's easier to recognize model name over hostname. Unless someone has two receivers of the same type, which seems like a very niche situation.